inference-engine/src/transformations/src/transformations/common_optimizations/stride_optimization.cpp

```diff
@@ -0,0 +1,198 @@
+    return it != rt_map.end();
+}
+
+static ngraph::Strides get_strides_prop(const ngraph::Input<ngraph::Node>& node) {
+    const auto& rt_map = node.get_rt_info();
+    const auto& var = rt_map.at(STRIDES_PROP_KEY);
+    return ngraph::as_type_ptr<ngraph::VariantWrapper<ngraph::Strides>>(var)->get();
+}
+
+static void insert_strides_prop(ngraph::Input<ngraph::Node> node, const ngraph::Strides& strides) {
+    auto& rt_map = node.get_rt_info();
+    rt_map[STRIDES_PROP_KEY] = std::make_shared<ngraph::VariantWrapper<ngraph::Strides>>(strides);
+}
+
+static bool can_propagate_conv_stride(const std::shared_ptr<ngraph::Node>& conv) {
+    const auto& kernel_pshape = conv->input_value(1).get_partial_shape();
+    if (kernel_pshape.is_dynamic())
+        return false;
+    const auto& kernel_shape = kernel_pshape.get_shape();
+    return std::all_of(kernel_shape.begin() + 2, kernel_shape.end(), [] (size_t s) -> bool { return s == 1; });
+}
+
+static std::vector<ngraph::Input<ngraph::Node>> get_node_target_inputs(const std::shared_ptr<ngraph::Node>& node) {
+    std::vector<ngraph::Input<ngraph::Node>> result;
+    for (auto output : node->outputs()) {
+        for (auto input : output.get_target_inputs()) {
+            result.push_back(input);
+        }
+    }
+    return result;
+}
+
+static std::tuple<std::vector<ngraph::Strides>, bool> check_next_ops(const std::vector<ngraph::Input<ngraph::Node>>& next_ops) {
+    std::vector<ngraph::Strides> strides;
+    for (const auto& op : next_ops) {
+        if (has_strides_prop(op)) {
+            strides.push_back(get_strides_prop(op));
+        }
+    }
+    bool all_ops_are_valid = !(next_ops.size() != strides.size() || (strides.size() > 0 &&
+                                                                     !std::all_of(strides.begin(), strides.end(),
+                                                                         [&strides] (const ngraph::Strides& s) -> bool {
+                                                                             bool all_ones = std::all_of(s.begin(), s.end(),
+                                                                                                         [] (size_t i) -> bool { return i == 1; });
+                                                                             return s == strides[0] && !all_ones;
+                                                                         })));
+    return std::make_tuple(strides, all_ops_are_valid);
+}
+
+static void insert_pooling(const std::shared_ptr<ngraph::Node>& first, ngraph::Input<ngraph::Node>& second, const ngraph::Strides& strides) {
+    auto pool = std::make_shared<ngraph::opset7::MaxPool>(first, strides, ngraph::Shape{}, ngraph::Shape{}, ngraph::Shape(strides.size(), 1));
+    second.replace_source_output(pool);
+}
+
+static void handle_not_equal_stride_props(const std::shared_ptr<ngraph::Node>& node, std::vector<ngraph::Input<ngraph::Node>>&& next_ops) {
```

Comment:
If node is a multioutput node then `insert_pooling` will fail because of node to output conversion.
I suggest to remove `node` argument from this function and use `input.get_source_output()` in order to get input for pooling operation.